src/runtime/hexagon/hexagon_wrapper_dl.h

```diff
@@ -0,0 +1,68 @@
+/*!
+ * Copyright 2019 Qualcomm Innovation Center, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE‐2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * This Contribution is being provided by Qualcomm Technologies, Inc.,
+ * a Delaware corporation, or its subsidiary Qualcomm Innovation Center, Inc.,
+ * a California corporation, under certain additional terms and conditions
+ * pursuant to Section 5 of the Apache 2.0 license.  In this regard, with
+ * respect to this Contribution, the term "Work" in Section 1 of the
+ * Apache 2.0 license means only the specific subdirectory within the TVM repo
+ * (currently at https://github.com/dmlc/tvm) to which this Contribution is
+ * made.
+ * In any case, this submission is "Not a Contribution" with respect to its
+ * permitted use with any of the "vta" and "verilog" subdirectories in the TVM
+ * repo.
+ */
+#ifndef TVM_RUNTIME_HEXAGON_HEXAGON_WRAPPER_DL_H_
+#define TVM_RUNTIME_HEXAGON_HEXAGON_WRAPPER_DL_H_
+
+#include "HexagonWrapper.h"
+
+struct HexagonWrapperDL {
+  HexagonWrapperDL(HEXAPI_Cpu cpu, HEX_4u_t flags = 0,
+                   HEX_4u_t recordflag = 0);
+  ~HexagonWrapperDL();
+
+  HEXAPI_Status ConfigureExecutableBinary(const char* fElf);
+  HEXAPI_Status EndOfConfiguration();
+  HEXAPI_Status LoadExecutableBinary(const char* fElf = nullptr);
+  HEXAPI_Status GetSimulatedCycleCount(HEX_8u_t* cycles);
+  HEXAPI_Status SetBreakpoint(HEX_VA_t virtAddr);
+  HEXAPI_Status ReadSymbolValue(const char* symbol, HEX_4u_t* value);
+  HEXAPI_Status ReadVirtual(HEX_VA_t virtAddr, HEX_4u_t asid, HEX_4u_t size,
+                            void* value);
+  HEXAPI_Status WriteVirtual(HEX_VA_t virtAddr, HEX_4u_t asid, HEX_4u_t size,
+                             HEX_8u_t value);
+
+  HEXAPI_CoreState Run(HEX_4u_t* result = nullptr);
+
+ private:
+  void* resolve(const char* sym_name);
+
+  void* sim_wrapper = nullptr;  // "Emulated" pointer to HexagonWrapper.
```

Comment:
google c convention, add underscore at the end of the variable.

Reply:
Done